File: app/src/main/java/crux/bphc/cms/models/core/UserDetail.java
package crux.bphc.cms.models.core;

import com.google.gson.annotations.SerializedName;

import crux.bphc.cms.network.MoodleServices;

/**
 * Model class to represent the response from
 * {@link MoodleServices#fetchUserDetail}
 *
 * @author Abhijeet Viswa
 */
public class UserDetail {
    @SerializedName("username") private String username;
    @SerializedName("firstname") private String firstName;
    @SerializedName("lastname") private String lastName;
    @SerializedName("userpictureurl") private String userPictureUrl;
    @SerializedName("userid") private int userId;

    private String token;

    public String getUsername() {
        return username;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getUserPictureUrl() {
        return userPictureUrl;
    }

    public int getUserId() {
        return userId;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }
}

File: app/src/main/java/crux/bphc/cms/models/core/UserDetail.kt
package crux.bphc.cms.models.core

import com.google.gson.annotations.SerializedName

/**
 * Model class to represent the response from [crux.bphc.cms.network.MoodleServices.fetchUserDetail]
 * @author Abhijeet Viswa
 */
data class UserDetail (
    @SerializedName("username") val username: String = "",
    @SerializedName("firstname")  val firstName: String = "",
    @SerializedName("lastname") val lastName: String = "",
    @SerializedName("userpictureurl") val userPictureUrl: String = "",
    @SerializedName("userid") val userId: Int = 0,
    var token: String = "",
)
